### PR TITLE
Improve cudnn performance when using deterministic mode

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
The test was successful.

File: cupy/cudnn.pyx [hunk 1]
```diff
@@ -1051,18 +1058,35 @@ cpdef _warn_algorithm_bwd_filter(
...
+            else:
+                raise RuntimeError(
+                    'No conv bwd filter algo available with workspace size '
+                    'less equal {}'.format(max_workspace_size))
+        else:
+            perf = cudnn.findConvolutionBackwardFilterAlgorithmEx_v7(
+                handle, x_desc, x.data.ptr, dy_desc, dy.data.ptr, conv_desc,
+                filter_desc, dW.data.ptr, 1, workspace.ptr,
+                max_workspace_size)[0]
+        algo = _get_algorithm(perf.algo, perf.memory, perf.mathType)
```
Comment (asi1024) [inline cupy/cudnn.pyx]: A compile error occurs: `undeclared name not builtin: _get_algorithm`.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

File: tests/cupy_tests/test_cudnn.py [hunk 2]
```diff
@@ -292,7 +292,7 @@ def setUp(self):
                 (self.groups > 1 and version < 7000)):
             self.err = ValueError
         elif deterministic and (
-                (self.dilate > 1 and (ndim != 2 or version < 7000)) or
+                (self.dilate > 1 and (ndim != 2 or version < 7300)) or
```
Comment (asi1024) [inline tests/cupy_tests/test_cudnn.py]: Could you add some comments about what is used in each version of cuDNN, and why the test has conditional branches by the version of cuDNN?

Comment (okuta) [reply tests/cupy_tests/test_cudnn.py]: I fixed.

Comment (okuta) [reply tests/cupy_tests/test_cudnn.py]: It is a little difficult because it is determined by cuDNN inner specification.

**Comment by asi1024** [comment]
Could you resolve conflicts?

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by okuta** [comment]
Jenkins, test this please.

**Comment by okuta** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM!

**Pull request merged**